### DISCUSSION TITLE
[20 BETA] Alphabetical order in launcher

### DISCUSSION THREAD

**Discussion by lxwndr** [linuxdeepin/developer-center#2983] [Features Request & Ideas | 特性请求 & 头脑风暴]
## Describe the bug
Allow the launcher icons to be arranged in alphabetical order. The current method is confusing and the user can get lost in the middle of so many icons.

The alphabetical order is already consolidated on _Android_ and _Microsoft Windows_. And if I'm not mistaken, also on the _Plasma Desktop_ (KDE) interface!

## Screenshots
My launcher in alphabetical order (it's even more beautiful!)

![Screen Capture_20200711222409](https://user-images.githubusercontent.com/45263160/87236906-4877b600-c3c5-11ea-920b-3dffa906fcb5.png)

### Distro & Version
deepin 20 BETA

### Additional context
If it is not possible to add this feature by default; please allow the user to choose between "confusing mode" (current) or alphabetic mode.

**Comment by XMuli** [comment]
There is no plan to change it for the time being, it may be considered in the future

**Comment by XMuli** [comment]
The current sort is probably more practical (in the previous version, I remember there was an a-z sort)

**Comment by lxwndr** [comment]
> There is no plan to change it for the time being, it may be considered in the future

Is it possible to add the 2 organization modes? In this case, the user can choose which one to use.

By clicking on the circular icon "A-z", the icons are automatically rearranged.

![Screen Capture_select-area_20200713235120](https://user-images.githubusercontent.com/45263160/87378253-cc67a480-c563-11ea-891a-01d0cf91a685.png)

_Note: For the Western, this organization is quite confusing. As I mentioned initially, there are several examples of consolidated systems that adopt the alphabetical order._

**Comment by acschill** [comment]
> The current sort is probably more practical (in the previous version, I remember there was an a-z sort)

Could you expand on how it's currently being sorted? Might make it easier to make sense of it when using..

**Comment by lxwndr** [comment]
The problem persists in the new version: deepin 20.1 (1010) / OsBuild=11031.108

**Comment by meiyixiang** [comment]
I have received this problem and will feedback it to relevant colleagues to see if there is any plan to support it

**Reply by yarecco** [reply]
This is the only linux distro in the whole universe that doesn't show applications in launcher  in alphabetical order or even allow to do that.
It is funny that developpers spend time to do useless stuff like rounded dock corners and do not focus on things that are user-friendly.
Deepin is good-looking but difficult to use as a daily driver.

**Reply by BLumia** [reply]
> developpers

Actually, designers :joy:

**Comment by meiyixiang** [comment]
我会努力去说服我们的产品添加此功能，抱歉给你带来的不便，可以先使用搜索查找应用
39/5000 
I will try my best to persuade our products to add this function. Sorry for the inconvenience brought to you. You can use the search application first

**Reply by yarecco** [reply]
Sure I can use search to run applications.
But the question is what is exactly this deepin launcher application made for? Killing time on searching apps on random list?

**Reply by BLumia** [reply]
He's trying to help you and provide a workaround answer to you and I guess saying mean words are not helpful at all :/

**Reply by yarecco** [reply]
Problem persists since the very beginning of the distro. It was raised here over 2 years ago.  Reply came only few days ago.
 What you mean "workaround"? Telling that I can use search sounds more mean to me.

**Reply by BLumia** [reply]
> It was raised here over 2 years ago. Reply came only few days ago.

We are sorry we didn't pay attention to user feedback from GitHub and we are working on this problem recently (see https://github.com/linuxdeepin/developer-center/issues/2831 for detail).

> What you mean "workaround"?

Before the product manager agrees to include such a feature in DDE, using the search method to locate application is the only possible way to achieve your goal.

> Telling that I can use search sounds more mean to me.

Document the request and pass it to relevant colleagues is all the help @meiyixiang can offer. I think he did his best now.

By the way. As a (currently) company driven product, it's the related project manager's decision to include a feature to the final project, and if they decide not to include a feature, they still don't owe you anything so there's nothing to blame. If you are not happy about their decision, you can also create a fork of the project and add the feature you need if you like. We're FOSS, isn't it?

As a free and open-source community, you could be a part of the community, helping implement such features and share it with other community members. Even if you can't code, leaving suggestions is still fine, just please try to be helpful :)

**Comment by lxwndr** [comment]
As I manifested myself in the initial topics; I think it is extremely important for deepin to adopt this organizational model. Currently, the app launcher is quite confusing!

The need to use the launcher's search to find any applications, contradicts the simplicity that the system aims to defend. It's one more step that the user doesn't want to do.

**Comment by meiyixiang** [comment]
If our product team doesn't accept this advice, I hope the community can do it themselves.
